src/tests/functional/plugin/shared/src/behavior/compiled_model/properties.cpp

```diff
@@ -0,0 +1,344 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "behavior/compiled_model/properties.hpp"
+#include "openvino/runtime/properties.hpp"
+#include <cstdint>
+
+namespace ov {
+namespace test {
+namespace behavior {
+
+std::string OVClassCompiledModelEmptyPropertiesTests::getTestCaseName(testing::TestParamInfo<std::string> obj) {
+    return "target_device=" + obj.param;
+}
+
+void OVClassCompiledModelEmptyPropertiesTests::SetUp() {
+    target_device = this->GetParam();
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+    APIBaseTest::SetUp();
+    model = ov::test::behavior::getDefaultNGraphFunctionForTheDevice(target_device);
+}
+
+std::string OVClassCompiledModelPropertiesTests::getTestCaseName(testing::TestParamInfo<PropertiesParams> obj) {
+    std::string targetDevice;
+    AnyMap properties;
+    std::tie(targetDevice, properties) = obj.param;
+    std::replace(targetDevice.begin(), targetDevice.end(), ':', '.');
+    std::ostringstream result;
+    result << "targetDevice=" << targetDevice << "_";
+    if (!properties.empty()) {
+        result << "properties=" << util::join(util::split(util::to_string(properties), ' '), "_");
+    }
+    return result.str();
+}
+
+void OVClassCompiledModelPropertiesTests::SetUp() {
+    std::tie(target_device, properties) = this->GetParam();
+    SKIP_IF_CURRENT_TEST_IS_DISABLED();
+    APIBaseTest::SetUp();
+    model = ov::test::behavior::getDefaultNGraphFunctionForTheDevice(target_device);
+}
+
+void OVClassCompiledModelPropertiesTests::TearDown() {
+    if (!properties.empty()) {
+        utils::PluginCache::get().reset();
+    }
+    APIBaseTest::TearDown();
+}
+
+// check properties
+std::string OVCompileModelGetExecutionDeviceTests::getTestCaseName(testing::TestParamInfo<OvPropertiesParams> obj) {
+    std::string target_device;
+    std::pair<ov::AnyMap, std::string> userConfig;
+    std::tie(target_device, userConfig) = obj.param;
+    std::replace(target_device.begin(), target_device.end(), ':', '.');
+    auto compileModelProperties = userConfig.first;
+    std::ostringstream result;
+    result << "device_name=" << target_device << "_";
+    if (!compileModelProperties.empty()) {
+        result << "_compileModelProp=" << util::join(util::split(util::to_string(compileModelProperties), ' '), "_");
+    }
+    result << "_expectedDevice=" << userConfig.second;
+    return result.str();
+}
+
+void OVCompileModelGetExecutionDeviceTests::SetUp() {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED();
+    std::pair<ov::AnyMap, std::string> userConfig;
+    std::tie(target_device, userConfig) = GetParam();
+    compileModelProperties = userConfig.first;
+    expectedDeviceName = userConfig.second;
+    model = ngraph::builder::subgraph::makeConvPoolRelu();
+}
+
+TEST_P(OVClassCompiledModelPropertiesTests, CanUseCache) {
+    core->set_property(ov::cache_dir("./test_cache"));
```

Comment:
move "./test_cache" to local var to avoid duplications

Reply:
done